rllib/evaluation/worker_set.py

```diff
@@ -267,9 +267,7 @@ def add_workers(self, num_workers: int, validate: bool = False) -> None:
         # which needs to be handled by this WorkerSet's owner (usually
         # a RLlib Algorithm instance).
         if validate:
-            self.foreach_worker_with_index(
-                lambda w, i: w.policy_map and w.input_reader and w.output_writer
-            )
+            self.foreach_worker(lambda w: w.assert_healthiness())
```

Comment:
I am curious, can you quickly try if it works to put assert in the lambda, not in the function directly?
then, we can make the function is_healthy(), and may potentially be more useful.

Reply:
Yup, just changed it, ready to merge

Reply:
Hmm, I'm okay either way, should we merge this? (Either before or after the last commit)

Reply:
Yeah, you can't use assert in a lambda. [Link](https://stackoverflow.com/questions/8477346/checking-assertions-in-a-lambda-in-python)

Still don't think that's the underlying issue, though, but I haven't been able to come up with an MWP that fails.

I added a test case to `test_io.py` to make sure it works correctly.